blockchain/types/tx_internal_data_access_list.go

```diff
@@ -311,6 +311,22 @@ func (t *TxInternalDataAccessList) SerializeForSign() []interface{} {
 	}
 }
 
+func (t *TxInternalDataAccessList) TxHash() common.Hash {
+	return prefixedRlpHash(byte(t.Type()), []interface{}{
+		t.ChainID,
```

Comment:
Do we need to set chainId of the transaction in the `func (s londonSigner) Hash(tx *Transaction) common.Hash` when chainId has nil or empty value?
If not set, signature can be invalid because chainId will be nil in TxHash but signature made with s.chainid i think?

Reply:
https://github.com/klaytn/klaytn/pull/1171/commits/b97c0192c7534b5d65fc3b7b41a3f661a7f4c0a9